config/state.go

```diff
@@ -35,19 +37,58 @@ type MixState struct {
 	/* hidden properties */
 	filename string
 	version  string
+
+	/* Inform the user where the source of mix format*/
+	formatSource string
 }
 
 // CurrentStateVersion is the current revision for the state file structure
 var CurrentStateVersion = "1.0"
 
 // LoadDefaults initialize the state object with sane values
 func (state *MixState) LoadDefaults() {
-	state.Mix.Format = "1"
+	state.loadDefaultFormat()
 
 	state.filename = "mixer.state"
 	state.version = CurrentStateVersion
 }
 
+func (state *MixState) loadDefaultFormat() {
+	/* Get format from legacy config file */
+	format, err := state.getFormatFromConfig()
+	if err == nil && format != "" {
+		state.Mix.Format = format
+		state.formatSource = "builder.conf"
+		return
+	}
+
+	/* Get format from system */
+	formatBytes, err := ioutil.ReadFile("/usr/share/defaults/swupd/format")
```

Comment:
Since this is configurable in swupd (`DEFAULT_FORMAT_PATH`) should it be configurable here? I know, yuck, but if it has potential to change...

Reply:
Ideally we should have something that keeps both projects in sync regarding this path, but I'm not sure having the same build option is the best way to go.

Reply:
OK then at least a const?

Reply:
Sounds good to me! I'll update the PR.